test/e2e/common/node/lease.go

```diff
@@ -89,7 +89,9 @@ var _ = SIGDescribe("Lease", func() {
 
 		readLease, err := leaseClient.Get(context.TODO(), name, metav1.GetOptions{})
 		framework.ExpectNoError(err, "couldn't read Lease")
-		framework.ExpectEqual(apiequality.Semantic.DeepEqual(lease.Spec, readLease.Spec), true)
+		if !apiequality.Semantic.DeepEqual(lease.Spec, readLease.Spec) {
+			framework.Failf("expected lease spec: %#v, got %#v", lease.Spec, readLease.Spec)
```

Comment:
We don't have good pretty-printing of API objects. Gomega's format.Object indents them nicely, but prints empty fields. There's a discussion going elsewhere about that.

I don't have a strong opinion about printf-style format strings. Can you perhaps try out which looks best (%s, %+v, %#v)?

Reply:
If we print as a single line, the printing the two objects on top of each other is easier to compare:
"Leases don't match. Expected:\n\t%s\nActual:\n\t%s"

Reply:
I wrote a piece of code to experiment:
```
s1 := spec{
		HolderIdentity:       sPtr("holder"),
		LeaseDurationSeconds: iPtr(30),
		AcquireTime:          &microTime{Time: time.Time{}.Add(2 * time.Second)},
		RenewTime:            &microTime{Time: time.Time{}.Add(5 * time.Second)},
		LeaseTransitions:     iPtr(0),
}
fmt.Printf("%#v\n", s1)
fmt.Println("=====================")
fmt.Printf("%s\n", s1)
fmt.Println("=====================")
fmt.Printf("%+v", s1)
```
And result is:
```
main.spec{HolderIdentity:(*string)(0xc000010250), LeaseDurationSeconds:(*int32)(0xc000018030), AcquireTime:time.Date(1, time.January, 1, 0, 0, 2, 0, time.UTC), RenewTime:time.Date(1, time.January, 1, 0, 0, 5, 0, time.UTC), LeaseTransitions:(*int32)(0xc000018034)}
=====================
{%!s(*string=0xc000010250) %!s(*int32=0xc000018030) 0001-01-01 00:00:02 +0000 UTC 0001-01-01 00:00:05 +0000 UTC %!s(*int32=0xc000018034)}
=====================
{HolderIdentity:0xc000010250 LeaseDurationSeconds:0xc000018030 AcquireTime:0001-01-01 00:00:02 +0000 UTC RenewTime:0001-01-01 00:00:05 +0000 UTC LeaseTransitions:0xc000018034}
```
I personally prefer the `%+v` output, informative enough and not too long. What do you think?

Reply:
I agree, let's use that.

Reply:
However, it won't help to diagnose the problem. It's better than before, but not good enough. For example, if HolderIdentity is different, then the output will just show the addresses, not the string.

Perhaps Failf should show the result of https://github.com/google/go-cmp?

Reply:
go-cmp would be much better there 👍

Reply:
I switched to use go-cmp.